### DISCUSSION TITLE
Two word only streaming detection. Like "Hey google", "Hey Siri"

### DISCUSSION THREAD

**Discussion by talkingtab** [openai/whisper#739] [Q&A]
I'm building a  voice note taker using Electron and webrtc. I would like to be able to detect a two word sequence, like "BANG BANG" in order to pause the streaming. Example: "I'm going to talk about webrtc today bang bang". The idea is to record everything (it could even include the bang bang), but then provide a signal for the application to pause. A person would resume the recording after that point by clicking a button. In general the complete transcription would be done after the fact, so the recognition during streaming is very, very limited.

What would be the best approach to this? Train a new model? Use something else? I'm literate in programming, not so much in AI.

This seems like a useful and common sort of thing, but maybe there is a better way?  
(Edit:  it would be good to have this run on an Android/IOS phone but would be useful even if that was not possible)

**Comment by silvacarl2** [comment]
yess if you combine real-time streaming with --initial-prompt, this can work.

**Reply by talkingtab** [reply]
I had not really thought this through very well. What I really want is to have whisper run fast enough to pause the recording process.  I ran whisper on a previous recording "A story about ducks bang bang specifically a duck named fred" with the --initial_prompt bang and was hoping I would get "bang specifically fred" but got the whole sentence. It appears from the doc that --initial_prompt reinforces the recognition of some words which does not really help my situation.

I think I can get some of what I want in post processing, embedding bang title all about ducks bang as I'm speaking, then extract the title in post processing. 

Whisper is just amazing!
